### PR TITLE
Rdi 50 update opensearch mappings

### PR DESCRIPTION
**Pull request by hakbailey**
#### What does this PR do?
This PR migrates the JSON generator and other ingest-related functions to match the new TIMDEX data model. Non-JSON generators are not being migrated. 

-  Updates the OpenSearch mappings document (config/es_record_mappings.json) and record model (pkg/record/record.go) to match the new TIMDEX data model
-  Adds a new fixture of TIMDEX JSON sample records for testing and updates the jsonrecord_test to match
-  Updates the elastic client to use the new TimdexRecordId field as the
-  OpenSearch document ID for each indexed record
-  Removes all of the non-JSON source generators and associated test files
-  Update Makefiles to use new AWS account and ECR repository name
  
#### Helpful background context
As we update the TIMDEX ETL process, we are splitting up the transform and load steps that have previously been coupled in this codebase. This PR updates the ingest code to *only* accept ingest data in the TIMDEX JSON format. Data is loaded directly into OpenSearch as TIMDEX JSON, and is not transformed here. New transform processes are  migrated to other codebase(s).

#### How can a reviewer manually see the effects of these changes?
Start a local OpenSearch Docker container as described in the readme. Install Mario with `make install`, then run `mario ingest -s mario --new fixtures/timdex_record_samples.json`. You should be able to query the OpenSearch instance in your browser at "localhost:9200/<index name>/_search?q=<query>" and see some of the indexed records.

#### Requires Full Reindexing of all Sources?
NO

#### Includes new or updated dependencies?
NO

#### Todo:
- [x] Tests
- [ ] Documentation
- [ ] Stakeholder approval

### CODE REVIEW

File: Makefile
Comment (zotoMIT) [inline Makefile:3]: I will update this after its merged, since i will likely need to make changes to this makefile anyway.   But we shouldn't hard code the account numbers here, its easier to manage if we use this method - https://github.com/MITLibraries/timdex-pipeline-lambdas/blob/995c43944db692e669f0ef91a9cd24b79087c3cb/Makefile#L3
`$(shell aws sts get-caller-identity --query Account --output text)`
and rely on the credentials to pull the value for the current account.

Comment (JPrevost) [reply Makefile:3]: Interesting. That does require that anyone running this to have the correct credentials for the stage they are trying to push to active. I agree with your suggestion here but I think we may need to talk through the workflow here as we are making subtle but important changes from when everything was in one account to splitting things across multiple accounts in terms of what will happen when commands are run and how to document that so dev expectations match reality so we don't accidentally push stuff to prod when we meant to stage them, etc.

**Review by JPrevost — COMMENTED** [review]
I'm good with the changes overall. I flagged a few places where I think we can delete more, and only half looked at the specific data mappings assuming if they are wrong we'll fix them as we add real data. If you'd prefer to remove the additional files under a separate ticket let me know and I'll approve once we make sure to open tickets to track that additional work.

File: pkg/record/record.go
Comment (hakbailey) [reply pkg/record/record.go:49]: I think we just wouldn't include that field if we don't know.

File: pkg/record/record.go
Comment (hakbailey) [reply pkg/record/record.go:54]: Yeah I think the API would have to handle querying both. Regardless, I'm not sure there's a better way to index the data since we know we have both single dates and date ranges.

File: pkg/record/record.go
Comment (JPrevost) [reply pkg/record/record.go:49]: I'm not fully convinced that a `bool` with a third state of undefined (being the most common variant) is the best option here from the perspective of the consumer of the API. I'm okay with seeing how this plays out and looking back at this field once we have more data in the system that attempts to populate it.

File: Makefile
Comment (zotoMIT) [reply Makefile:3]: We need to talk about how we want the promotion process to go for sure.  You're correct @JPrevost that this would require correct credentials for anything to happen, and it will always pick the active credentials account to attempt to push to.

**Review by JPrevost — APPROVED** [review]
👍🏻

**Pull request merged**
